app/main/presenters/search_presenters.py

```diff
@@ -182,21 +188,24 @@ def build_lots_and_categories_link_tree(framework, lots, category_filter_group,
     Returns a list of selected filters: there will always be at least one element, which is
     the root of the tree; there may then be a category; there may then follow a sub-category.
 
-    :param framework: the latest G-Cloud framework from the API
+    :param framework: the framework from the API
     :param lots: a sequence of lot dicts applicable to the live framework(s)
     :param category_filter_group: a single filter group loaded from the framework search_filters manifest
-    :param request: current request so that we can figure out what lots and categories are selected
-    :param content_manifest: a ContentManifest instance for G-Cloud search_filters.
-    :return: list of selected category and lot filters, starting with the 'all categories' root node node
+    :param request: the original request for creating links for nodes
+    :param cleaned_request_args: current cleaned request args so that we can figure out selected lots and categories
+    :param content_manifest: a ContentManifest instance for the frameworks search_filters.
+    :param search_link_builder: a werkzeug Href object for the view category tree links link to
+    :return: list of selected category and lot filters, starting with the 'all categories' root node
     """
-    current_lot_slug = get_lot_from_args(request.args, [lot['slug'] for lot in lots])
+    current_lot_slug = get_lot_from_args(cleaned_request_args, [lot['slug'] for lot in lots])
 
     # Links in the tree should preserve all the filters, except those relating to this tree (i.e. lot
     # and category).
-    search_link_builder = Href(url_for('.search_services'))
     keys_to_remove = _get_category_filter_key_set(category_filter_group)
     keys_to_remove.add('page')
-    preserved_request_args = MultiDict((k, v) for (k, v) in request.args.items(multi=True) if k not in keys_to_remove)
+    preserved_request_args = MultiDict(
+        (k, v) for (k, v) in request.args.items(multi=True) if k not in keys_to_remove
+    )
 
     selected_filters = list()
     # Create root node for the tree, always selected, which is the parent of the various lots.
```

Comment:
Don't know why we're building this dictionary instead of just creating it.
I feel like someing along the lines of:
```
    root_node = {
        'label': 'All categories',
        'selected': True,
        'link': search_link_builder(_build_base_url_args(preserved_request_args, content_manifest, framework, None)),
        'children': []
    }
```

Would let the reader see an obvious example of a node before diving in to the construction of the child elements below.

Reply:
Good shout, have updated.

Reply:
Have done the same for the lot_filters just below.